vault/expiration.go

```diff
@@ -422,6 +480,88 @@ func (m *ExpirationManager) Restore(errorFunc func()) (retErr error) {
 	m.restoreModeLock.Unlock()
 
 	m.logger.Info("lease restore complete")
+
+	if atomic.LoadUint32(&m.SHA1HashedLeasesCleared) == 0 {
+		expMetadata := &ExpirationMetadata{}
+		if atomic.LoadUint32(&m.SHA1HashedUnleasedTokensUpgraded) == 0 {
```

Comment:
I think we want to run this upgrade even if the `SHA1HashedLeasesCleared == 1` That way we can guarantee the root tokens are always upgraded

Reply:
Do you see a possibility of SHA1HashedUnleasedTokensUpgraded to be 0 when SHA1HashedLeasesCleared is 1? I am under the impression that when SHA1HashedLeasesCleared is 1, there is a guarantee that all the root tokens are upgraded.